Sources/joyfill-components-swift/Components/ComponentViewController/APIComponents/API/JoyfillAPI.swift

```diff
@@ -261,8 +301,30 @@ func getChartAndImageValue(valueElements: [ValueElement], j: Int) {
         for k in 0..<valueElements.count {
             pickedSingleImg = [valueElements[k].url ?? ""]
             pickedImg.append(valueElements[k].url ?? "")
-            DispatchQueue.main.async {
-                componentTableView.reloadData()
+        }
+    }
+    
+    optionsData = joyFillStruct?.fields?[j].tableColumns ?? []
+    for item in valueElements {
+        if (item.deleted ?? false) == false {
+            valueData.append(item)
+        }
+    }
+    for rows in 0..<(joyFillStruct?.fields?[0].rowOrder?.count ?? 0) {
+        tableRowOrder.append(joyFillStruct?.fields?[0].rowOrder?[rows] ?? "")
+    }
+    for columns in 0..<(joyFillStruct?.fields?[0].tableColumnOrder?.count ?? 0) {
```

Comment:
I would suggest we add code comments to each one of these for loops. I would suggest this for the future as well. New devs coming in might struggle to understand it

Reply:
Done